frontend/app/src/components/hexagonalBoard/HexagonalBoard.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+	useState,
+	ReactNode,
+	useEffect,
+	forwardRef,
+	useImperativeHandle,
+} from "react";
+import HexagonalTile, {
+	HexagonalTileProps,
+} from "@/components/hexagonalTile/HexagonalTile.tsx";
+import "./HexagonalBoard.css";
+
+const defaultTileProps: HexagonalTileProps = {
+	fill: "",
+	isDimmed: false,
+	stroke: "gray",
+	strokeWidth: "16",
+	identifier: "",
+	textContent: "",
+};
+
+interface HexagonalBoardProps {
+	rows: number;
+	tilesPerRow: number;
+	ref?: React.Ref<{
+		updateTile: (
+			row: number,
+			col: number,
+			props: Partial<HexagonalTileProps>,
+		) => void;
+	}>;
+}
+
+const HexagonalBoard = forwardRef(function HexagonalBoard(
+	{ rows, tilesPerRow }: HexagonalBoardProps,
+	ref,
+) {
+	const [tileProps, setTileProps] = useState<Map<string, HexagonalTileProps>>(
+		new Map(),
+	);
+
+	useEffect(() => {
+		const initialTileProps = new Map<string, HexagonalTileProps>();
+		for (let i = 0; i < rows; i++) {
+			for (let j = 0; j < tilesPerRow; j++) {
+				const tileId = `tile-${i}-${j}`;
+				initialTileProps.set(tileId, {
+					fill: "",
+					isDimmed: false,
+					stroke: "gray",
+					strokeWidth: "16",
+					identifier: tileId,
+					textContent: "",
```

Comment:
Would there be a clean way to use the `defaultTileProps`  values and only edit the tileId?
Maybe even a function `getTileProps` with optional arguments, so you'd just have to say `getTileProps(tileId=tileId)`?

Reply:
maybe something like
{
    ...defaultTileProps,
    identifier: tileId,
} could also work

Reply:
switched to the suggestion of @Straczi